controller/node_controller.go

```diff
@@ -1120,6 +1121,35 @@ func (nc *NodeController) syncMultipathd(node *longhorn.Node, namespaces []lhtyp
 	node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeMultipathd, longhorn.ConditionStatusTrue, "", "")
 }
 
+func (nc *NodeController) checkModulesLoaded(node *longhorn.Node, namespaces []lhtypes.Namespace) {
+	checkingModules := []string{
+		"dm_crypt",
+	}
+	nsexec, err := lhns.NewNamespaceExecutor(lhtypes.ProcessNone, lhtypes.HostProcDirectory, namespaces)
+	if err != nil {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeModulesLoaded, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonNamespaceExecutorErr),
+			fmt.Sprintf("Failed to get namespace executor: %v", err.Error()))
+		return
+	}
+
+	notLoadedModules := []string{}
+	for _, module := range checkingModules {
+		args := []string{module, "/proc/modules"}
+		if _, err := nsexec.Execute(nil, "grep", args, lhtypes.ExecuteDefaultTimeout); err != nil {
+			notLoadedModules = append(notLoadedModules, module)
+		}
```

Comment:
err might be not a `NotFound` error.

Reply:
Do you think it is enough only to add a `Debug` log to check what kind of error occurred as what `syncPackagesInstalled` does?

The debug log was added. Thanks

Reply:
I mean the error might be `not loaded`. The current implementation makes all errors become not loaded error and might make users confused.